lib/react_on_rails/version_checker.rb

```diff
@@ -2,44 +2,49 @@ module ReactOnRails
   # Responsible for checking versions of rubygem versus yarn node package
   # against each otherat runtime.
   class VersionChecker
-    attr_reader :node_package_version, :logger
-    MAJOR_VERSION_REGEX = /(\d+)\.?/
+    attr_reader :node_package_version
+    MAJOR_MINOR_PATCH_VERSION_REGEX = /(\d+)\.(\d+)\.(\d+)/
 
     def self.build
-      new(NodePackageVersion.build, Rails.logger)
+      new(NodePackageVersion.build)
     end
 
-    def initialize(node_package_version, logger)
-      @logger = logger
+    def initialize(node_package_version)
       @node_package_version = node_package_version
     end
 
     # For compatibility, the gem and the node package versions should always match,
     # unless the user really knows what they're doing. So we will give a
     # warning if they do not.
-    def warn_if_gem_and_node_package_versions_differ
+    def raise_if_gem_and_node_package_versions_differ
       return if node_package_version.relative_path?
-      return if node_package_version.major == gem_major_version
-      log_differing_versions_warning
+      node_major_minor_patch = node_package_version.major_minor_patch
+      gem_major_minor_patch = gem_major_minor_patch_version
+      return if node_major_minor_patch[0] == gem_major_minor_patch[0] &&
+                node_major_minor_patch[1] == gem_major_minor_patch[1] &&
+                node_major_minor_patch[2] == gem_major_minor_patch[2]
+
+      raise_differing_versions_warning
     end
 
     private
 
-    def log_differing_versions_warning
-      msg = "**WARNING** ReactOnRails: ReactOnRails gem and node package MAJOR versions do not match\n" \
+    def raise_differing_versions_warning
+      msg = "**ERROR** ReactOnRails: ReactOnRails gem and node package versions do not match\n" \
```

Comment:
Ok, So I think I mis-read the diff at ln29:33, but there is no-change at ln: 32:36 that I think should change to read `"Ensure the installed version of the gem is the same as the version of \n"\`
At Changelog.md:2, maybe should say `NOTE: exact versions of the npm module and the gem must be kept in sync.`